bindings/gumjs/gumquickmemory.c

```diff
@@ -931,21 +932,63 @@ GUMJS_DEFINE_FUNCTION (gumjs_memory_scan)
   GumMemoryScanContext sc;
   gpointer address;
   gsize size;
-  const gchar * match_str;
+  JSValue pattern_val;
+  JSValue callbacks;
+
+  if (args->count < 4)
+    return _gum_quick_throw_literal (ctx, "missing argument");
+
+  if (!_gum_quick_native_pointer_get (ctx, args->elements[0], core,
+      &address))
+    return JS_EXCEPTION;
 
-  if (!_gum_quick_args_parse (args, "pZsF{onMatch,onError?,onComplete}",
-      &address, &size, &match_str, &sc.on_match, &sc.on_error, &sc.on_complete))
+  if (!_gum_quick_size_get (ctx, args->elements[1], core, &size))
     return JS_EXCEPTION;
+
+  pattern_val = args->elements[2];
+
+  if (JS_IsString (pattern_val)) {
```

Comment:
```suggestion
  if (JS_IsString (pattern_val))
  {
```

(Here and elsewhere.)